org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/analysis/IJythonFunction.java

```diff
@@ -0,0 +1,13 @@
+package org.eclipse.scanning.sequencer.analysis;
+
+import org.eclipse.january.dataset.IDataset;
+
+public interface IJythonFunction {

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Annotate the "IJythonFunction" interface with the @FunctionalInterface annotation (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1609)